packages/service-provider-core/package-lock.json

```diff
@@ -313,24 +337,18 @@
       "optional": true
     },
     "node-abi": {
-      "version": "2.30.1",
-      "resolved": "https://registry.npmjs.org/node-abi/-/node-abi-2.30.1.tgz",
-      "integrity": "sha512-/2D0wOQPgaUWzVSVgRMx+trKJRC2UG4SUc4oCJoXx9Uxjtp0Vy3/kt7zcbxHF8+Z/pK3UloLWzBISg72brfy1w==",
+      "version": "3.8.0",
+      "resolved": "https://registry.npmjs.org/node-abi/-/node-abi-3.8.0.tgz",
+      "integrity": "sha512-tzua9qWWi7iW4I42vUPKM+SfaF0vQSLAm4yO5J83mSwB7GeoWrDKC/K+8YCnYNwqP5duwazbw2X9l4m8SC2cUw==",
       "optional": true,
       "requires": {
-        "semver": "^5.4.1"
+        "semver": "^7.3.5"
       }
     },
     "node-addon-api": {
-      "version": "4.2.0",
-      "resolved": "https://registry.npmjs.org/node-addon-api/-/node-addon-api-4.2.0.tgz",
-      "integrity": "sha512-eazsqzwG2lskuzBqCGPi7Ac2UgOoMz8JVOXVhTvvPDYhthvNpefx8jWD8Np7Gv+2Sz0FlPWZk0nJV0z598Wn8Q==",
-      "optional": true
-    },
-    "noop-logger": {
-      "version": "0.1.1",
-      "resolved": "https://registry.npmjs.org/noop-logger/-/noop-logger-0.1.1.tgz",
-      "integrity": "sha1-lKKxYzxPExdVMAfYlm/Q6EG2pMI=",
+      "version": "4.3.0",
```

Comment:
The e2e test failures are happening because of this – the node-addon-api `Error` class constructor isn’t ABI-compatible between 4.3.0 and previous versions due to this addition: https://github.com/nodejs/node-addon-api/blame/2c88a7ec4c6d14a27182f5de6a5166faa9e86567/napi.h#L1723

Since in the compiled binary multiple node-addon-api instances from different addons end up being linked together, that can end up being problematic (in this case, the kerberos addon ends up using the newer, incompatible node-addon-api from mongodb-client-encryption for creating Error objects, but uses inline code from the older node-addon-api for destroying them). I’ve opened a PR for node-addon-api to undo this incompatibility, since there was never a good reason for adding it to begin with, but obviously that won’t be released automatically either.

I’m not sure what the best approach here is. https://github.com/mongodb-js/kerberos/pull/142 will update the dependency for kerberos in the same way.

Ideally, we probably need to find a way to avoid these conflicts altogether. For now, maybe we could try to standardize on node-addon-api 4.3.0 across the board? #1205 updates macos-export-certificate-and-key/win-export-certificate-and-key to versions that also use that node-addon-api version.

Reply:
Got it, thanks for the detailed explanation! Aligning them sounds like a good idea, from what I can see in addition to these certificate packages we would also need to bump `os-dns-native`, right? (here's an npm ls node-addon-api from mongosh):

```
mongosh: mongosh@0.0.0-dev.0 /Users/sergey.petushkov/Projects/MongoDB/mongosh/packages/mongosh
mongosh: └─┬ @mongosh/cli-repl@0.0.0-dev.0 -> ./../cli-repl
mongosh:   ├─┬ @mongosh/logging@0.0.0-dev.0 -> ./../logging
mongosh:   │ └─┬ @mongodb-js/devtools-connect@1.1.3
mongosh:   │   └─┬ os-dns-native@1.1.2
mongosh:   │     └── node-addon-api@3.2.1
mongosh:   ├─┬ @mongosh/service-provider-core@0.0.0-dev.0 -> ./../service-provider-core
mongosh:   │ └─┬ mongodb-client-encryption@2.0.0
mongosh:   │   └── node-addon-api@4.3.0
mongosh:   ├─┬ @mongosh/service-provider-server@0.0.0-dev.0 -> ./../service-provider-server
mongosh:   │ ├─┬ @mongodb-js/devtools-connect@1.1.3
mongosh:   │ │ └─┬ os-dns-native@1.1.2
mongosh:   │ │   └── node-addon-api@3.2.1
mongosh:   │ ├─┬ kerberos@2.0.0-beta.0
mongosh:   │ │ └── node-addon-api@4.2.0
mongosh:   │ └─┬ mongodb-client-encryption@2.0.0
mongosh:   │   └── node-addon-api@4.3.0
mongosh:   ├─┬ @mongosh/types@0.0.0-dev.0 -> ./../types
mongosh:   │ └─┬ @mongodb-js/devtools-connect@1.1.3
mongosh:   │   └─┬ os-dns-native@1.1.2
mongosh:   │     └── node-addon-api@3.2.1
mongosh:   └─┬ macos-export-certificate-and-key@1.0.2
mongosh:     └── node-addon-api@1.7.2
```

Reply:
Yeah, I think we'll need to bump the node-addon-api version there before this (but I would expect the upgrade to Just Work 🙂)

Reply:
Seems like it indeed "just worked" 😄 https://github.com/mongodb-js/os-dns-native/pull/4